env0/resource_variable_set.go

```diff
@@ -0,0 +1,451 @@
+package env0
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var variableSetVariableSchema *schema.Resource = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"name": {
+			Type:        schema.TypeString,
+			Description: "variable name",
+			Required:    true,
+		},
+		"value": {
+			Type:        schema.TypeString,
+			Description: "variable value for 'hcl', 'json', or 'text' format",
+			Optional:    true,
+		},
+		"dropdown_values": {
+			Type:        schema.TypeList,
+			Description: "a list of variable values for 'dropdown' format",
+			Optional:    true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+		"type": {
+			Type:             schema.TypeString,
+			Description:      "variable type: terraform or environment (defaults to 'environment')",
+			Default:          "environment",
+			Optional:         true,
+			ValidateDiagFunc: NewStringInValidator([]string{"environment", "terraform"}),
+		},
+		"is_sensitive": {
+			Type:        schema.TypeBool,
+			Description: "is the value sensitive (defaults to 'false'). Note: 'dropdown' value format cannot be senstive.",
+			Optional:    true,
+			Default:     false,
+		},
+		"format": {
+			Type:             schema.TypeString,
+			Description:      "the value format: 'text' (free text), 'dropdown' (dropdown list), 'hcl', 'json'. Note: 'hcl' and 'json' can only be used in terraform variables.",
+			Optional:         true,
+			Default:          "text",
+			ValidateDiagFunc: NewStringInValidator([]string{"text", "dropdown", "hcl", "json"}),
+		},
```

Comment:
is there a way not to duplicate it between all the usages, environment / regular variable??

not need to be in this PR just for the future

Reply:
The current configuration variables in the environment has more features, but that's not the problem...
The implementation is lacking (bugs/limitations) and the schema is confusing (doesn't align with the UI).
But yes, once we add variables_sets to environment,  common code will be used.

About environment configuration - we could deprecate current implementation... and replace it  but that's a separate (large) effort...